File: src/mcp/search_quad.ts
import { QdrantClient } from "@qdrant/js-client-rest";
import dotenv from "dotenv";


dotenv.config();


const qdrant = new QdrantClient({
  url: process.env.QDRANT_URL!,   // from .env
  apiKey: process.env.QDRANT_KEY! // from .env
});

export async function searchContext(vector: number[]): Promise<string> {
  const result = await qdrant.search("document-qna", {
    vector,
    top: 4
  });

  const context = result
    .map((point) => point.payload?.text ?? "")
    .filter(Boolean)
    .join("\n\n");

  return context;
}

File: src/mcp/search_quad.py
import os
from qdrant_client import QdrantClient


def search_context(vector: list[float]) -> str:
    """Search Qdrant for context based on a vector."""
    qdrant_url = os.getenv("QDRANT_URL")
    qdrant_key = os.getenv("QDRANT_KEY")

    qdrant = QdrantClient(url=qdrant_url, api_key=qdrant_key)

    results = qdrant.query_points(
        collection_name="document-qna",
        query=vector,
        limit=5
    ).points

    context = "\n\n".join(
        point.payload.get("text", "") for point in results if point.payload
    )

    return context
